internal/devserver/coordinator.go

```
@@ -0,0 +1,163 @@
+	if options.Listener == nil {
+		return fmt.Errorf("serve.listener_required: listener is required")
+	}
+	if options.Builder == nil {
+		_ = options.Listener.Close()
+		return fmt.Errorf("serve.builder_required: builder is required")
+	}
+	if options.Changes == nil {
+		_ = options.Listener.Close()
+		return fmt.Errorf("serve.watcher_required: change source is required")
+	}
+	defer options.Changes.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `options.Changes.Close` is not checked

(errcheck)